### ISSUE TITLE
Cannot change the default text editor (Notepad.exe) to gVim.exe

### ISSUE THREAD

**Issue by 10p10110111xxi11p00100100** [ipython/ipython#6803]
In `ipython_qtconsole_config.py` none of these three lines start gVim.exe from IPython QT Console by typing out `%ed`:
```python
c.IPythonWidget.editor = u"C:/Program Files (x86)/Vim/vim74/gVim.exe"
c.IPythonWidget.editor = r"C:\Program Files (x86)\Vim\vim74\gVim.exe"
c.IPythonWidget.editor = "C:\\Program Files (x86)\\Vim\\vim74\\gVim.exe"
```
This is a similar issue as #3047.

System information:
{'commit_hash': 'be0cc2b',
 'commit_source': 'installation',
 'default_encoding': 'cp775',
 'ipython_path': 'C:\\Anaconda\\lib\\site-packages\\IPython',
 'ipython_version': '2.3.0',
 'os_name': 'nt',
 'platform': 'Windows-7-6.1.7601-SP1',
 'sys_executable': 'C:\\Anaconda\\python.exe',
 'sys_platform': 'win32',
 'sys_version': '2.7.6 |Anaconda 2.0.0 (32-bit)| (default, May 27 2014, 15:05:55) [MSC v.1500 32 bit (Intel)]'}

**Comment by r-learner** [comment]
Same issue. using ipython qtconsole, %edit does not work as expected. After saving the code, the code is not run. Without using qtconsole, %edit just works.

**Comment by Carreau** [comment]
> Same issue. using ipython qtconsole, %edit does not work as expected. After saving the code, the code is not run. Without using qtconsole, %edit just works.
> The OS is mac os x, the editor is macvim.

This is not the same issue. The "does not run code" after saving is normal, as with edit, the code is run when the editor program does exit and allow the python process to continue running. which with gui editor (macvim, or notepad) is not possible because of event loop. You can sometime go around that by passing some flag to your editor. (like -w for text mate).

**Comment by ellisonbg** [comment]
Do we need to do anything on this one?

**Comment by Gorticide** [comment]
When I was testing out the %edit function in IPython in both Linux and Windows, I was not pleased with the default text editors, vi in Linux, notepad in Windows.  

In Linux I edited the .bashrc file:

export EDITOR=emacs

In Windows I went to System | Advanced System Settings | Environment Variables.
I created system variable EDITOR and set the value to where GNU emacs is:  c:\tools\emacs\bin\emacs.exe

Both systems require a reboot before it takes effect.

I was going to post this discovery on my message board (whybother.freeboards.org), but nobody really cares about such matters there, so I figured I would place it here.

The solution* to edit the config file would be a good way to go, but I wanted to make it system-wide.

*
c.IPythonWidget.editor = u'"C:/Program Files (x86)/Vim/vim74/gVim.exe"'
